### PR TITLE
add sourcesets for godot-library and godot headers as submodule

### PR DESCRIPTION
**Pull request by piiertho**
This add sourcesets for godot-library, so that everyone can start to work.
This also add godot-headers submodule.
This makes #87 outdated.

### CODE REVIEW

File: godot-kotlin/godot-library/build.gradle.kts [hunk 1]
```diff
@@ -6,13 +6,19 @@ plugins {
...
-    val nativeMain = sourceSets.create("nativeMain")
+    macosX64("macos")
+    linuxX64("linux")
+    mingwX64("windows")
 
     targets.withType<KotlinNativeTarget> {
-        compilations.getByName("main").defaultSourceSet { dependsOn(nativeMain) }
+        compilations.getByName("main") {
+            defaultSourceSet {
+                kotlin.srcDirs("src/main/nativeGen/kotlin", "src/main/nativeInternal/kotlin", "src/main/nativeCore/kotlin", "src/main/nativePublic/kotlin")
```
Comment (raniejade) [inline godot-kotlin/godot-library/build.gradle.kts]: can we create the sourceSets via `sourceSets.create` then use dependsOn here?

**Review by raniejade — CHANGES_REQUESTED** [review]
One comment.

File: godot-kotlin/godot-library/build.gradle.kts [hunk 2]
```diff
@@ -6,13 +6,31 @@ plugins {
...
+    linuxX64("linux")
+    mingwX64("windows")
 
-    val nativeMain = sourceSets.create("nativeMain")
+    val internal = sourceSets.create("nativeInternal")
+    val generated = sourceSets.create("nativeGen")
+    val core = sourceSets.create("nativeCore")
+    val public = sourceSets.create("nativePublic")
+
+    core.dependsOn(internal)
```
Comment (raniejade) [inline godot-kotlin/godot-library/build.gradle.kts]: you can make it more concise by:

```kotlin
 	val internal = sourceSets.create("nativeInternal")
    val core = sourceSets.create("nativeCore") { dependsOn(internal }
    val generated = sourceSets.create("nativeGen") { dependsOn(core) }
    val public = sourceSets.create("nativePublic") { dependsOn(generated) }
```

**Review by raniejade — CHANGES_REQUESTED** [review]
Just one more thing, then LGTM!

**Review by raniejade — APPROVED** [review]

File: godot-kotlin/godot-library/build.gradle.kts [hunk 3]
```diff
@@ -6,13 +6,28 @@ plugins {
...
 //TODO: this needs to be properly configured! This is just a basic setup to be able to implement the annotations
 kotlin {
-    linuxX64()
-    mingwX64()
-    macosX64()
+    macosX64("macos")
+    linuxX64("linux")
+    mingwX64("windows")
+
+    val internal = sourceSets.create("nativeInternal")
```
Comment (chippmann) [inline godot-kotlin/godot-library/build.gradle.kts]: please don't just use the word `internal`. That's a keyword ;-)
Make it internalSourceSet or nativeInternal or internalNative :-)

File: godot-kotlin/godot-library/build.gradle.kts [hunk 4]
```diff
@@ -6,13 +6,28 @@ plugins {
...
+    val core = sourceSets.create("nativeCore") { dependsOn(internal) }
+    val generated = sourceSets.create("nativeGen") { dependsOn(core) }
+    val public = sourceSets.create("nativePublic") { dependsOn(generated) }
```
Comment (chippmann) [inline godot-kotlin/godot-library/build.gradle.kts]: same with public

**Review by chippmann — CHANGES_REQUESTED** [review]
Other fthan my comments, LGTM :-) Thanks a lot!

File: godot-kotlin/godot-library/build.gradle.kts (hunk 4 above)
Comment (piiertho) [reply godot-kotlin/godot-library/build.gradle.kts]: oh yes, morning things xD

**Review by chippmann — APPROVED** [review]

**Review by CedNaru — APPROVED** [review]

**Pull request merged**